Fix mobile spacing for case studies

deliaconstantino opened the pull request:
## Fix mobile styling on `case-study` pages 

1. Update `table` scroll on `pages/case-study/extrahop/index.mdx` to address mobile spacing bug
- Fixes gravitational/web#2759
- Preview `/extrahop/` [here](https://teleport-docs-next-git-delia-fix-case-stud-ab88ab-gravitational.vercel.app/case-study/extrahop/)

2. Add responsive styling for `<Review />` component.
- Preview `/auth0/` [here](https://teleport-docs-next-git-delia-fix-case-stud-ab88ab-gravitational.vercel.app/case-study/auth0/)
**Mobile size -** before on left; after on right:
<img width="1907" alt="Screen Shot 2022-02-14 at 5 33 52 PM" src="https://user-images.githubusercontent.com/70108137/153957908-b3b24a07-9d84-4c1f-9ca4-9a666ae89f60.png">


**Tablet size -** before on left; after on right:
<img width="1920" alt="Screen Shot 2022-02-14 at 5 37 00 PM" src="https://user-images.githubusercontent.com/70108137/153958286-d65c3d46-b174-4c40-a520-6909466b5465.png">

C-STYR commented:
Nice job, @deliaconstantino.  Let's see if we can fix the table in a different way; this may not be possible but we're going to try before we go with a scroll option (which is a bit clunky).  See if you can figure out how to make the columns resize appropriately so it all fits in 375px.   I see at least one problem with an item that isn't going to want to get smaller but you can probably modify the content slightly to get this to work or use some lesser-known CSS.  Don't hesitate to create a custom component on this page to accomplish either goal rather than inline styling.

deliaconstantino commented:
> Nice job, @deliaconstantino. Let's see if we can fix the table in a different way; this may not be possible but we're going to try before we go with a scroll option (which is a bit clunky). See if you can figure out how to make the columns resize appropriately so it all fits in 375px. I see at least one problem with an item that isn't going to want to get smaller but you can probably modify the content slightly to get this to work or use some lesser-known CSS. Don't hesitate to create a custom component on this page to accomplish either goal rather than inline styling.

- Created a custom `<CaseStudyTable />` component for these styles, nested inside a new directory called `StyledTable`. There already is a `Table` directory, but it seems to serve a different purpose than what we want here.

- Updated both tables in `/pages/case-study/extrahop/` to `<CaseStudyTable />`
<img width="983" alt="Screen Shot 2022-02-16 at 11 27 22 AM" src="https://user-images.githubusercontent.com/70108137/154310638-96b17ab1-5b3e-47b4-9e73-d85b98af4f62.png">


- Modified content slightly to ensure text wrapping at most appropriate spots
<img width="1425" alt="Screen Shot 2022-02-16 at 11 18 31 AM" src="https://user-images.githubusercontent.com/70108137/154308792-972cee6d-dc24-4505-bab1-9d74fac0df7f.png">

- Additionally, updated code spacing inside `<thead/>` to address/remove `invalidateDOMnesting(...)` warning from React:
<img width="1440" alt="Screen Shot 2022-02-16 at 10 57 01 AM" src="https://user-images.githubusercontent.com/70108137/154308323-b098fee8-5bc9-4750-80fa-f1fa227291c4.png">
